### PR TITLE
Measurement in raster result descriptor

### PR DESCRIPTION
**Pull request by ChristianBeilschmidt**

### CODE REVIEW

**Comment by ChristianBeilschmidt** [comment]
[Units von Layern anfragen/mitliefern](https://trello.com/c/TvGznR4U/274-units-von-layern-anfragen-mitliefern)

File: operators/src/engine/operator.rs [hunk 1]
```diff
@@ -195,3 +195,13 @@ impl From<Box<InitializedRasterOperator>> for TypedInitializedOperator {
         TypedInitializedOperator::Raster(operator)
     }
 }
+
+#[macro_export]
+macro_rules! call_on_typed_operator {
```
Comment (jdroenner) [inline operators/src/engine/operator.rs:200]: i thought we already have a macro like this

File: operators/src/processing/expression.rs [hunk 2]
```diff
@@ -70,21 +69,52 @@ impl RasterOperator for Expression {
                 found: self.vector_sources.len()
             }
         );
+        ensure!(
+            !self.raster_sources.is_empty(),
+            crate::error::InvalidNumberOfVectorInputs {
```
Comment (jdroenner) [inline operators/src/processing/expression.rs:74]: RasterInputs?

File: operators/src/processing/expression.rs [hunk 3]
```diff
@@ -70,21 +69,52 @@ impl RasterOperator for Expression {
...
+                    expected: spatial_reference,
+                    found: other_spatial_refenence,
+                }
+            );
+        }
+
+        let result_descriptor = RasterResultDescriptor {
+            data_type: self.params.output_type,
+            spatial_reference,
+            measurement: raster_sources[0].result_descriptor().measurement.clone(), // TODO: use parameter to define it
```
Comment (jdroenner) [inline operators/src/processing/expression.rs:105]: please do like output_type

File: operators/src/source/gdal_source.rs [hunk 4]
```diff
@@ -285,7 +287,7 @@ where
             .time_step
             .snap_relative(time_information.start_time, time_interval.start())
             .expect("is a valid time");
-        eprintln!("snapped start {:?}", snapped_start);
```
Comment (jdroenner) [inline operators/src/source/gdal_source.rs:288]: is there a lint against "print" in the code? :dagger:

File: services/src/handlers/workflows.rs [hunk 5]
```diff
@@ -71,16 +72,21 @@ async fn get_workflow_metadata<C: Context>(
...
-    let operator = operator
-        .initialize(&ctx.execution_context()?)
-        .context(error::Operator)?;
+    let execution_context = ctx.execution_context()?;
 
     // TODO: use cache here
-    let result_descriptor = operator.result_descriptor();
-
-    Ok(warp::reply::json(result_descriptor))
+    call_on_typed_operator!(
```
Comment (jdroenner) [inline services/src/handlers/workflows.rs:78]: do we really need the macro here? I fear that it somehow hides the initialize step and we should not initialize workflows more then needed. I know it i needed at the moment.

**Review by jdroenner — CHANGES_REQUESTED** [review]
i guess it should work. we can change the measurement if we need to,
Please check the error type in the expression operator.

File: operators/src/engine/operator.rs (hunk 1 above)
Comment (ChristianBeilschmidt) [reply operators/src/engine/operator.rs:200]: Where?

File: operators/src/processing/expression.rs (hunk 3 above)
Comment (ChristianBeilschmidt) [reply operators/src/processing/expression.rs:105]: This breaks the expression (UI, etc.), so the task for it is in the expression story.

File: operators/src/source/gdal_source.rs (hunk 4 above)
Comment (ChristianBeilschmidt) [reply operators/src/source/gdal_source.rs:288]: https://rust-lang.github.io/rust-clippy/master/index.html#print_stderr

File: services/src/handlers/workflows.rs (hunk 5 above)
Comment (ChristianBeilschmidt) [reply services/src/handlers/workflows.rs:78]: The macro doesn't cover the initialization, you see it in the function where it is used. So I can't share this fear.

File: operators/src/processing/expression.rs (hunk 2 above)
Comment (ChristianBeilschmidt) [reply operators/src/processing/expression.rs]: 112a95d

File: operators/src/processing/expression.rs (hunk 3 above)
Comment (ChristianBeilschmidt) [reply operators/src/processing/expression.rs]: d2c23f1

File: operators/src/engine/operator.rs (hunk 1 above)
Comment (jdroenner) [reply operators/src/engine/operator.rs:200]: i don't know. i just thought i have seen one just like this

File: operators/src/source/gdal_source.rs (hunk 4 above)
Comment (jdroenner) [reply operators/src/source/gdal_source.rs]: we should use it

File: services/src/handlers/workflows.rs (hunk 5 above)
Comment (jdroenner) [reply services/src/handlers/workflows.rs:88]: it's ok like this

**Pull request merged**